bundles/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/discovery/BigAssFanDiscoveryService.java

```diff
@@ -115,21 +115,25 @@ protected void stopBackgroundDiscovery() {
         cancelListenerJob();
     }
 
-    private void startListenerJob() {
-        if (listenerJob == null) {
-            terminate = false;
-            logger.debug("Starting discovery listener job in {} seconds", BACKGROUND_DISCOVERY_DELAY);
-            listenerJob = scheduledExecutorService.schedule(listenerRunnable, BACKGROUND_DISCOVERY_DELAY,
+    private synchronized void startListenerJob() {
```

Comment:
Does it really need to be synchronized? I would propose to follow the common pattern of using a local variable.

Reply:
I think it is needed here. As checking for null (even on a local var) and if null scheduling a task might cause multiple task to be scheduled. For the other way around (safely accessing a property) i agree.
When looking for `synchronized void start` i found many bindings scheduling a task like this. There are also more bindings that have the `synchronised` block on this within the method, so there might be many more bindings.